src/App.jsx

```diff
@@ -14,33 +16,118 @@ export const goodsFromServer = [
+    if (arr[i] !== ENTER_GOODS[i]) {
+      return false;
+    }
+  }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  return true;
+}
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+function getPreparedGoods(goods, sortField, reverse) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_NAME:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [firstList, nextList] = useState(goodsFromServer);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const [reverse, setReverse] = useState(false);
+  const visibleGoods = getPreparedGoods(firstList, sortField, reverse);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          onClick={() => {
+            nextList(ENTER_GOODS);
+            setSortField(SORT_FIELD_NAME);
+          }}
+          className={cn('button is-info', {
+            'is-light': sortField !== SORT_FIELD_NAME,
+          })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          onClick={() => {
+            nextList(ENTER_GOODS);
+            setSortField(SORT_FIELD_LENGTH);
+          }}
+          className={cn('button is-success', {
+            'is-light': sortField !== SORT_FIELD_LENGTH,
+          })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          onClick={() => {
+            setSortField(sortField);
+            // eslint-disable-next-line no-unused-expressions
+            !reverse ? setReverse(true) : setReverse(false);
+          }}
+          className={cn('button is-warning', {
+            'is-light': reverse === false,
+          })}
+        >
+          Reverse
+        </button>
+
+        {!filter(visibleGoods) && (
+          <button
+            type="button"
+            onClick={() => {
+              setSortField('');
+              nextList(ENTER_GOODS);
+              setReverse(false);
+            }}
+            className={cn('button is-danger is-light')}
+          >
+            Reset
+          </button>
+        )}
+      </div>
+
+      <div>
+        <ul>
+          {visibleGoods.map(good => (
+            <li data-cy="Good">{good}</li>
```

Comment:
As per the checklist, you need to add a `key` to all iterations. Each child in a list should have a unique 'key' prop. You can use the index of the map function as a key, but it's better to use unique and stable keys if you have ones. In this case, since all goods are unique, you can use the good itself as a key.